### PR TITLE
make task setting keys into an enum

### PR DESCRIPTION
**Pull request by meganrogge**

### CODE REVIEW

File: src/vs/workbench/contrib/tasks/common/tasks.ts [hunk 1]
```diff
@@ -1190,6 +1190,23 @@ export namespace KeyedTaskIdentifier {
...
+	Tasks = 'tasks',
+	SuppressTaskName = 'tasks.suppressTaskName',
+	Windows = 'tasks.windows',
+	Osx = 'tasks.osx',
+	Linux = 'tasks.linux',
+	ShowOutput = 'tasks.showOutput',
+	IsShellCommand = 'tasks.isShellCommand',
+	ServiceTestSetting = 'tasks.service.testSetting',
+}
+
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/common/tasks.ts:1209]: I notice some settings aren't there:

Screen Shot 2022-07-06 at 6 48 02 pm

Mean to include these too? https://github.com/Microsoft/vscode/blob/7c78640d86e5de1ca0270912584d9d38beafeec1/src/vs/workbench/contrib/tasks/browser/task.contribution.ts#L428

File: src/vs/workbench/contrib/tasks/common/tasks.ts [hunk 2]
```diff
@@ -1190,6 +1190,23 @@ export namespace KeyedTaskIdentifier {
 	}
 }
 
+export const enum TaskSettingId {
+	AutoDetect = 'task.autoDetect',
+	SaveBeforeRun = 'task.saveBeforeRun',
+	ShowDecorations = 'task.showDecorations'
+}
+
+export const enum TasksSettingId {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/common/tasks.ts:1199]: I'm not sure what this is exactly, it is properties in the tasks schema?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/tasks/common/tasks.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/tasks/common/tasks.ts:1209]: maybe should add those too - but i was just going for the ones that are referenced within the code base

Comment (Tyriar) [reply src/vs/workbench/contrib/tasks/common/tasks.ts:1209]: :+1: including everything means there's a single source of truth and strong typing where it's used

File: src/vs/workbench/contrib/tasks/common/tasks.ts (hunk 2 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/tasks/common/tasks.ts:1199]: yes json schema v2 which is unfortunate bc don't think we can use these types as the keys there bc they're localized

Comment (meganrogge) [reply src/vs/workbench/contrib/tasks/common/tasks.ts:1199]: so maybe not worth doing this for those

Comment (Tyriar) [reply src/vs/workbench/contrib/tasks/common/tasks.ts:1199]: Yeah can't use in localize ids. How about naming this `TasksSchemaProperties`?

**Review by Tyriar — APPROVED** [review]

**Pull request merged**